miqa/core/models/evaluation.py

```diff
@@ -5,7 +5,12 @@
 
 class Evaluation(models.Model):
     id = models.UUIDField(primary_key=True, default=uuid4, editable=False)
-    frame = models.ForeignKey('Frame', null=False, on_delete=models.PROTECT)
+    frame = models.OneToOneField(
+        'Frame',
+        null=False,
+        on_delete=models.PROTECT,
```

Comment:
Why `PROTECT` and not `CASCADE`?

Reply:
perhaps I have their behavior reversed, but I thought that the desired outcome was that deleting an Evaluation would not delete the target Frame, but deleting a Frame would delete all related Evaluations. Which is best to use for that?

Reply:
When you set `on_delete` on Django ForeignKeys, it roughly mirrors the semantics of `ON DELETE` on the RDBMS foreign keys, meaning it refers to what happens when the _referenced_ object is deleted. So in this case, if you want deleting a Frame to delete all associated Evaluations, you want `models.CASCADE`. In the other direction, nothing will happen regardless of this value.

* `PROTECT` would mean that an attempt to delete the associated Frame would raise an exception and fail.
* `SET_NULL` would mean that deleting a Frame would preserve all the evaluations that were linked to it, but set their `frame` column to null.

Reply:
I see, thanks for the clarification. I think I always get them switched because I think of what happens when the _defined_ object is deleted. I'll make that change.